ios/Exponent/Versioned/Core/Api/EXNotifications.m

```diff
@@ -92,41 +93,116 @@ - (instancetype)initWithExperienceId:(NSString *)experienceId kernelServiceDeleg
+      } else {
+        resolve(content.userInfo[@"id"]);
+      }
+    }];
   }];
 
-  resolve(notification.userInfo[@"id"]);
+}
+
+RCT_EXPORT_METHOD(addCategory: (NSString *) categoryId
+                  actions: (NSArray *) actions
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(__unused RCTPromiseRejectBlock)reject)
+{
+  NSMutableArray<UNNotificationAction *> * actionsArray = [[NSMutableArray alloc] init];
+  
+  for( NSArray * action in actions) {
+    int optionsInt = [(NSNumber *)action[2] intValue];
+    UNNotificationActionOptions options = UNNotificationActionOptionForeground + optionsInt;
+                                     
+    if (([action count] == 5)) {
+      UNTextInputNotificationAction * newAction = [UNTextInputNotificationAction actionWithIdentifier:action[0]
+                                                                                                title:action[1]
+                                                                                              options:options
+                                                                                 textInputButtonTitle:action[3] textInputPlaceholder:action[4]];
+      [actionsArray addObject:newAction];
+    } else {
+      UNNotificationAction * newAction = [UNNotificationAction actionWithIdentifier:action[0] title:action[1] options:options];
+      [actionsArray addObject:newAction];
+    }
+  }
+  
+  UNNotificationCategory * newCategory = [UNNotificationCategory categoryWithIdentifier:categoryId actions:actionsArray intentIdentifiers:@[] options:UNNotificationCategoryOptionNone];
+  
+  [[UNUserNotificationCenter currentNotificationCenter] setNotificationCategories:[NSSet setWithObject:newCategory]];
+  resolve(@"done");
 }
 
 RCT_EXPORT_METHOD(scheduleLocalNotification:(NSDictionary *)payload
                   withOptions:(NSDictionary *)options
                   resolver:(RCTPromiseResolveBlock)resolve
                   rejecter:(__unused RCTPromiseRejectBlock)reject)
 {
-  UILocalNotification *notification = [self _localNotificationFromPayload:payload];
-
-  notification.fireDate = [RCTConvert NSDate:options[@"time"]] ?: [NSDate new];
-  notification.repeatInterval = [RCTConvert NSCalendarUnit:options[@"repeat"]] ?: 0;
-
+  bool repeats = NO;
+  if (options[@"repeats"]) {
+    repeats = ([options[@"repeats"] isEqualToString:@"Yes"])? YES : NO;
+  }
+  UNMutableNotificationContent* content = [self _localNotificationFromPayload:payload];
+  
+  NSDateComponents * date = [[NSDateComponents alloc] init];
+  NSArray * unites = @[@"day", @"month", @"year", @"weekday", @"quarter", @"leapMonth", @"nanosecond", @"era", @"weekdayOrdinal", @"weekOfMonth", @"weekOfYear", @"hour", @"second", @"minute", @"yearForWeekOfYear"];
+  for( NSString * unit in unites) {
+    if (options[unit]) [date setValue:(NSNumber *)options[unit] forKey:unit];
```

Comment:
Maybe we should stick to supporting `repeat` https://docs.expo.io/versions/v30.0.0/sdk/notifications#repeat-optional-string----minute-hour-day?